fdbserver/KeyValueStoreRocksDB.actor.cpp

```diff
@@ -244,41 +371,155 @@ struct RocksDBKeyValueStore : IKeyValueStore {
 
 		struct OpenAction : TypedAction<Writer, OpenAction> {
 			std::string path;
+			std::string dataPath;
 			ThreadReturnPromise<Void> done;
 			Optional<Future<Void>>& metrics;
 			const FlowLock* readLock;
 			const FlowLock* fetchLock;
-			OpenAction(std::string path,
+			KeyRangeMap<DB>* shardMap;
```

Comment:
Consider encapsulating path and DB with a struct, to avoid duplicate KeyRangeMap.